staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -167,3 +167,41 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32 `json:"samplingRatePerMillion,omitempty" protobuf:"varint,2,opt,name=samplingRatePerMillion"`
 }
+
+// FlowControlConfig holds priority and fairness parameters.
+type FlowControlConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// WorkEstimatorConfig holds work estimator parameter values.
+	*WorkEstimatorConfig `json:"workEstimatorConfig,omitempty"`
+}
+
+// WorkEstimatorConfig holds work estimator parameters.
+type WorkEstimatorConfig struct {
+	ListWorkEstimatorConfig     *ListWorkEstimatorConfig     `json:"listWorkEstimatorConfig,omitempty"`
+	MutatingWorkEstimatorConfig *MutatingWorkEstimatorConfig `json:"mutatingWorkEstimatorConfig,omitempty"`
+
+	// MinimumSeats is the minimum number of seats a request must occupy.
+	MinimumSeats uint `json:"minimumSeats,omitempty"`
+	// MaximumSeats is the maximum number of seats a request can occupy
+	//
+	// NOTE: work_estimate_seats_samples metric uses the value of maximumSeats
+	// as the upper bound, so when we change maximumSeats we should also
+	// update the buckets of the metric.
+	MaximumSeats uint `json:"maximumSeats,omitempty"`
+}
+
+// ListWorkEstimatorConfig holds work estimator parameters related to list requests.
+type ListWorkEstimatorConfig struct {
+	ObjectsPerSeat float64 `json:"objectsPerSeat,omitempty"`
+}
+
+// MutatingWorkEstimatorConfig holds work estimator
+// parameters related to watches of mutating objects.
+type MutatingWorkEstimatorConfig struct {
+	// TODO(wojtekt): Remove it once we tune the algorithm to not fail
+	// scalability tests.
+	Enabled                 bool            `json:"enable,omitempty"`
```

Comment:
@wojtek-t : is this still a question?  I thought the scalability tests have long been passing now.

Reply:
The upstream ones are passing but we had some issues with other kinds of tests internally. So I would like to leave this option for a bit.

Reply:
Leaving.